src/spatial_reaction_network.jl

```diff
@@ -0,0 +1,109 @@
+# A network of compartments (a meta graph that can contain some additional infro for each compartment).
+# The lattice is a DiGraph, normals graphs are converted to DiGraphs (with one edge in each direction).
+struct LatticeReactionSystem # <: MT.AbstractTimeDependentSystem # Adding this part messes up show, disabling me from creating LRSs
+    """The spatial reactions defined between individual nodes."""
+    rs::ReactionSystem
+    """The spatial reactions defined between individual nodes."""
+    spatial_reactions::Vector{SpatialReaction}
+    """The graph on which the lattice is defined."""
+    lattice::DiGraph
+    """Dependent (state) variables representing amount of each species. Must not contain the
+    independent variable."""
+    
+    function LatticeReactionSystem(rs, spatial_reactions, lattice::DiGraph)
+        return new(rs, spatial_reactions, lattice)
+    end
+    function LatticeReactionSystem(rs, spatial_reactions, lattice::SimpleGraph)
+        return new(rs, spatial_reactions, DiGraph(lattice))
+    end
+end
+
+### ODEProblem ###
+# Creates an ODEProblem from a LatticeReactionSystem.
+function DiffEqBase.ODEProblem(lrs::LatticeReactionSystem, u0, tspan,
+                               p = DiffEqBase.NullParameters(), args...;
+                               kwargs...)
+    @unpack rs, spatial_reactions, lattice = lrs       
+                        
+    spatial_params = unique(getfield.(spatial_reactions, :rate))
+    pV_in, pE_in = split_parameters(p, spatial_params)
+    nV, nE = length.([vertices(lattice), edges(lattice)])
+    u_idxs = Dict(reverse.(enumerate(Symbol.(getfield.(states(rs), :f)))))
+    pV_idxes = Dict(reverse.(enumerate(Symbol.(parameters(rs)))))
+    pE_idxes = Dict(reverse.(enumerate(unique(getfield.(spatial_reactions, :rate)))))
+
+    u0 = matrix_form(u0, nV, u_idxs)
+    pV = matrix_form(pV_in, nV, pV_idxes)
+    pE = matrix_form(pE_in, nE, pE_idxes);
+
+    return ODEProblem(build_f(lrs, u_idxs, pE_idxes), u0, tspan, (pV,pE), args...; kwargs...)
+end
+
+# Splits parameters into those for the compartments and those for the connections.
+split_parameters(parameters::Tuple, spatial_params) = parameters
+split_parameters(parameters::Vector, spatial_params) = filter(p -> !in(p[1], spatial_params), parameters), filter(p -> in(p[1], spatial_params), parameters);
+
+# Converts species and parameters to matrices form.
+matrix_form(input::Matrix, args...) = input
+matrix_form(input::Vector{Pair{Symbol,Vector{Float64}}}, n, index_dict) = mapreduce(permutedims, vcat, last.(sort(input, by=i -> index_dict[i[1]])))
+matrix_form(input::Vector, n, index_dict) = matrix_form(map(i -> (i[2] isa Vector) ? i[1] => i[2] : i[1] => fill(i[2], n), input), n, index_dict);
+
+# Creates a function for simulating the spatial ODE with spatial reactions.
+function build_f(lrs::LatticeReactionSystem, u_idxs::Dict{Symbol,Int64}, pE_idxes::Dict{Symbol,Int64})

```

Comment:
That's probably the way to go, I'm just unfamiliar with actually using the approach and its advantages,

Reply:
https://docs.julialang.org/en/v1/manual/methods/#Function-like-objects

```julia
mutable struct GraphODEFunWrapper
   p1
   p2
end

# in-place ODE derivative function
function (g::GraphODEFunWrapper)(du, u, p, t)
    # can now use g.p1 and g.p2 
end
```
So you create a new `GraphODEFunWrapper` to store the needed parameters that aren't in `p` and return that to the user. It can still be called like a function in the ODE solvers.